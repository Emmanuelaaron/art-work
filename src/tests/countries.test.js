import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import countryReducer, { initialState } from '../redux/countries/countries';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});

describe('country reducer', () => {
  it('should return the initial state', () => {
    expect(countryReducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle LOAD_DATA', () => {
    expect(
      countryReducer(initialState, {
        type: 'LOAD_DATA',
      }),
    ).toMatchSnapshot();
  });
  it('should handle COUNTRY_CLICKED', () => {
    expect(
      countryReducer(initialState, {
        type: 'COUNTRY_CLICKED',
      }),
    ).toMatchSnapshot();
  });
  it('should handle FILTER_COUNTRY', () => {
    expect(
      countryReducer(initialState, {
        type: 'FILTER_COUNTRY',
      }),
    ).toMatchSnapshot();
  });
});

describe('async actions', () => {
  const fetchCountriesData = {
    dates: {
      '2020-12-12': {
        countries: {
          uganda: {
            name: 'uganda',
            'name-es': 'uganda',
            today_confirmed: 9,
            regions: [],
          },
        },
      },
    },
  };

  it('dispatches LOAD_DATA when fetching countries has been done', () => {
    store.dispatch({ type: 'LOAD_DATA' }, fetchCountriesData);
    expect(store.getActions()).toMatchSnapshot();
  });
});
