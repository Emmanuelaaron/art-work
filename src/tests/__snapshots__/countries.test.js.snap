// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async actions dispatches LOAD_DATA when fetching countries has been done 1`] = `
Array [
  Object {
    "type": "LOAD_DATA",
  },
]
`;

exports[`country reducer should handle COUNTRY_CLICKED 1`] = `
Object {
  "countries": Array [],
  "detail": "",
  "filtererdCountries": Array [],
}
`;

exports[`country reducer should handle FILTER_COUNTRY 1`] = `
Object {
  "countries": Array [],
  "detail": "",
  "filtererdCountries": Array [],
}
`;

exports[`country reducer should handle LOAD_DATA 1`] = `
Object {
  "countries": Array [],
  "detail": "",
  "filtererdCountries": Array [],
}
`;

exports[`country reducer should return the initial state 1`] = `
Object {
  "countries": Array [],
  "detail": "",
  "filtererdCountries": Array [],
}
`;
